salt/utils/templates.py

```diff
@@ -99,6 +97,54 @@ def __getattr__(self, name):
 
 
 def wrap_tmpl_func(render_str):
+    def _sls_context(tmplpath, sls):
+        """
+        Generate SLS/Template Context Items
+        """
+
+        sls_context = {}
+
+        # Normalize SLS as path
+        slspath = sls.replace(".", "/")
+
+        if tmplpath is not None:
+            # Normalize template path
+            template = tmplpath.replace("\\", "/")
+
+            # Determine proper template name without root
+            if template.endswith("{}.sls".format(slspath)):
+                template = template[-(4 + len(slspath)) :]
+            elif template.endswith("{}/init.sls".format(slspath)):
+                template = template[-(9 + len(slspath)) :]
+            else:
+                # Something went wrong
+                log.warning("Failed to determine proper template path")
```

Comment:
Generally speaking, I bet `pathlib.Path` would be useful here. It already contains methods for renaming, getting the parent, etc. - it would make the logic throughout this function much more apparent.

Reply:
Is there any scenario under which the sls would not match the template path name? This else is a bit of my paranoia, but I am not sure if it is ever going to be reached. I will check out `pathlib.Path` though. If there is a better way than comparing sls to template path, I'd rather do that and not have all these conditions

Reply:
I looked into pathlib - but I am not sure how it is useful here. The issue is that if sls does not map to template path, there is no way to figure out which part of the template path is the "file root" and which is the template within said root. I am not sure any standard library would be able to do this as this is something only meaningful within the code. I am not sure if there is ever a scenario where it would fall through to the `else` section, so it is mostly there for defensive purposes in case either I do not understand things correctly or something changes in the future.